src/kzg.rs

```diff
@@ -75,56 +68,24 @@ fn verifier(proof: Proof, z: Fr, trusted_setup: TrustedSetup) -> bool {
 #[cfg(test)]
 mod tests {
     use crate::kzg::{prover, verifier, Fr};
-    use crate::kzg_tools::{
-        barycentric::barycentric,
-        fft::fft,
-        polynomial::{self, div_poly, mul_poly},
-        trusted_setup::trusted_setup,
-    };
+    use crate::kzg_tools::polynomial::Evaluation;
+    use crate::kzg_tools::{polynomial::Polynomial, trusted_setup::trusted_setup};
     use halo2::arithmetic::Field;
+    use halo2::halo2curves::ff::PrimeField;
     use rand::thread_rng;
 
     #[test]
     fn kzg_test() {
-        let p_committed = polynomial::Polynomial::new(vec![1, 2, 3, 4, 5, 6, 7, 8]);
-        let rou = p_committed.clone().rou();
-        let p_fr = &p_committed.p_to_fr();
-
-        //Evaluate a polynomial
-        println!("{:?}\n", &p_committed.eval(16));
-
-        //Evaluate a polynomial without FFT
-        println!("{:?}\n", &p_committed.eval_fr(rou));
-
-        //Evaluate a polynomial using FFT
-        println!("{:?}\n", fft(p_fr.clone(), rou));
-
-        let p_numerator = polynomial::Polynomial::new(vec![3, 2, 1, 6, 7, 9]);
-        let p_denominator = polynomial::Polynomial::new(vec![2, 1, 5, 6]);
-        println!("{} ÷ {}\n", p_numerator, p_denominator);
-
-        //Divide two polynomials.
-        //Numerator = 3x^0 + 2x^1 + 1x^2 denominator = 2x^0 + 1x^1 quotient = 0 + 1x and remainder = 3
-        println!(
-            "{:?}\n",
-            div_poly(p_numerator.p_to_fr(), p_denominator.p_to_fr())
-        );
-
-        //Multiplying two polynomials.
-        //First polynomial = 3x^0 + 2x^1 + 1x^2 second polynomial = 2x^0 + 1x^1 result = 6 + 7x + 4x^2 + 1x^3
-        println!(
-            "{:?}\n",
-            mul_poly(p_numerator.p_to_fr(), p_denominator.p_to_fr())
-        );
-
-        //Evaluate polynomial using barycentric
-        println!("{:?}\n", barycentric(p_fr.clone(), rou, 16.into()));
-
-        //KZG commitment
+        let p_committed = Polynomial::new(vec![
+            Fr::from_u128(16),
+            Fr::from_u128(2),
+            Fr::from_u128(5),
+            Fr::from_u128(12),
+        ]);
         let rng = thread_rng();
         let z = Fr::random(rng.clone());
-        let trusted_setup = trusted_setup(p_fr.to_vec());
-        let prover = prover(p_committed, z, trusted_setup.clone());
+        let trusted_setup = trusted_setup(p_committed.clone());
+        let prover = prover(p_committed, Evaluation::new(z), trusted_setup.clone());
```

Comment:
Evalutation::new(z) is not correct. z is the point that we evaluate and get the y point. That means z is not an evaluation point it is the evaluated point